### PR TITLE
For #1063: Put tab tray behind a feature flag and hidden pref

### PR DESCRIPTION
**Pull request by boek**

### CODE REVIEW

**Comment by darkwing** [comment]
Took an early peek at this quickly.  Found something:

- The "Open Tabs" header displays when there are no open tabs and the pref is off.  It think we still need the empty message and collections:

Screenshot_20200430-130211

**Comment by darkwing** [comment]
I also cause a crash when I click the private browsing button on the top right of the home screen:

```
2020-04-30 13:43:17.267 21370-21370/org.mozilla.fenix.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.fenix.debug, PID: 21370
    java.lang.IllegalStateException: Fragment HomeFragment{34872ca} (8a74e32e-d0ba-4376-b768-89f196279993)} not attached to a context.
        at androidx.fragment.app.Fragment.requireContext(Fragment.java:805)
        at org.mozilla.fenix.home.HomeFragment$onCreateView$15.invoke(HomeFragment.kt:247)
        at org.mozilla.fenix.home.HomeFragment$onCreateView$15.invoke(HomeFragment.kt:111)
        at mozilla.components.lib.state.ext.ViewKt$consumeFrom$1.invokeSuspend(View.kt:37)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
```

This would be with the pref off and secret settings off.

**Comment by darkwing** [comment]
Code looks good, couldn't find any UI problems.  R+!  Excellent!

File: app/src/main/java/org/mozilla/fenix/settings/SecretSettingsPreference.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import android.os.Bundle
+import androidx.preference.PreferenceFragmentCompat
+import androidx.preference.SwitchPreference
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.getPreferenceKey
+import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.ext.showToolbar
+
+/**
+ * Lets the user customize Private browsing options.
```
Comment (cadeyrn) [inline app/src/main/java/org/mozilla/fenix/settings/SecretSettingsPreference.kt:16]: @boek The comment is wrong (copied from PrivateBrowsingFragment, I guess?)

**Pull request merged**